packages/react-aria-components/src/Accordion.tsx

```diff
@@ -0,0 +1,150 @@
+  /**
+   * State of the accordion item.
+   */
+  state: DisclosureState
+}
+
+export const AccordionItemContext = createContext<ContextValue<AccordionItemProps, HTMLDivElement>>(null);
+export const DisclosureStateContext = createContext<DisclosureState | null>(null);
+
+interface InternalAccordionContextValue {
+  contentProps: DOMAttributes<HTMLElement>,
+  contentRef: React.RefObject<HTMLElement | null>
+}
+
+const InternalAccordionContext = createContext<InternalAccordionContextValue | null>(null);
+
+function AccordionItem(props: AccordionItemProps, ref: ForwardedRef<HTMLDivElement>) {
+  [props, ref] = useContextProps(props, ref, AccordionItemContext);
+  let state = useDisclosureState(props);
+  let contentRef = useRef<HTMLElement>(null);
+  let {triggerProps, contentProps} = useDisclosure({...props, contentRef}, state);
+  let {
+    isFocusVisible: isFocusVisibleWithin,
+    focusProps: focusWithinProps
+  } = useFocusRing({within: true});
+
+  let renderProps = useRenderProps({
+    ...props,
+    defaultClassName: 'react-aria-AccordionItem',
+    values: {
+      isExpanded: state.isExpanded,
+      isDisabled: props.isDisabled || false,
+      isFocusVisibleWithin,
+      state
+    }
+  });
+
+  return (
+    <Provider
+      values={[
+        [ButtonContext, {
+          slots: {
+            trigger: triggerProps
+          }
+        }],
+        [InternalAccordionContext, {contentProps, contentRef}],
+        [DisclosureStateContext, state]
+      ]}>
+      <div
+        ref={ref}
+        data-expanded={state.isExpanded || undefined}
+        data-disabled={props.isDisabled || undefined}
+        data-focus-visible-within={isFocusVisibleWithin || undefined}
+        {...focusWithinProps}
+        {...renderProps}>
+        {renderProps.children}
+      </div>
+    </Provider>
+  );
+}
+
+export interface AccordionPanelProps extends RenderProps<{}> {
+  /**
+   * The accessibility role for the accordion item's panel.
+   * @default 'group'
+   */
+  role?: 'group' | 'region',
+  children: ReactNode
+}
+
+function AccordionPanel(props: AccordionPanelProps, ref: ForwardedRef<HTMLElement>) {
+  let {role = 'group'} = props;
+  let {contentProps, contentRef} = useContext(InternalAccordionContext)!;
+  let {
+    isFocusVisible: isFocusVisibleWithin,
+    focusProps: focusWithinProps
+  } = useFocusRing({within: true});
+  let renderProps = useRenderProps({
+    ...props,
+    defaultClassName: 'react-aria-AccordionPanel',
+    values: {
+      isFocusVisibleWithin
+    }
+  });
+  let mergedRef = useObjectRef(mergeRefs(contentRef, ref));
+  return (
+    <div
+      role={role}
+      // @ts-ignore
+      ref={mergedRef}
+      {...mergeProps(contentProps, focusWithinProps)}
+      {...renderProps}
+      data-focus-visible-within={isFocusVisibleWithin || undefined}>
+      <Provider
+        values={[
+          [ButtonContext, null]
+        ]}>
+        {props.children}
+      </Provider>
+    </div>
+  );
+}
+
+/**
```

Comment:
We should probably have an `Accordion` component in RAC at some point as well to support mutually exclusive accordions.